tsc/src/video/img_set.hpp

```diff
@@ -0,0 +1,211 @@
+/***************************************************************************
+ * img_set.h
+ *
+ * Copyright © 2003 - 2011 Florian Richter
+ * Copyright © 2013 - 2014 The TSC Contributors
+ ***************************************************************************/
+/*
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 3 of the License, or
+   (at your option) any later version.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef TSC_IMG_SET_HPP
+#define TSC_IMG_SET_HPP
+
+#include "../core/global_game.hpp"
+#include "../core/global_basic.hpp"
+#include "../core/math/rect.hpp"
+#include "../core/file_parser.hpp"
+#include "../video/video.hpp"
+
+namespace TSC {
+    /* *** *** *** *** *** *** *** cImageSet_FrameInfo *** *** *** *** *** *** *** *** *** *** */
+    struct cImageSet_FrameInfo {
+        cImageSet_FrameInfo();
+
+        boost::filesystem::path m_filename;
+        Uint32 m_time_min;
+        Uint32 m_time_max;
+
+        typedef std::pair<int, int> Entry_Type;
+        typedef std::vector<Entry_Type> List_Type;
+
+        List_Type m_branches;
+    };
+
+    /* *** *** *** *** *** *** *** cImageSet_Parser *** *** *** *** *** *** *** *** *** *** */
+    class cImageSet_Parser : public cFile_parser
+    {
+    public:
+        typedef std::vector<cImageSet_FrameInfo> List_Type;
+
+        cImageSet_Parser(Uint32 time);
+        bool HandleMessage(const std::string* parts, unsigned int count, unsigned int line);
+
+        List_Type m_images;
+        Uint32 m_time_min;
+        Uint32 m_time_max;
+    };
+
+    /* *** *** *** *** *** *** *** cImageSet_Surface *** *** *** *** *** *** *** *** *** *** */
+
+    class cImageSet_Surface {
+    public:
+        cImageSet_Surface(void);
+        ~cImageSet_Surface(void);
+
+        // enter the frame
+        void Enter(void);
+        // leave a frame, return next frame for branching or -1 
+        int Leave(void);
+
+        // the image
+        cGL_Surface* m_image;
+        // time to display in milliseconds
+        Uint32 m_time;
+        // information
+        cImageSet_FrameInfo m_info;
+    };
+
+
+    /* *** *** *** *** *** *** *** cImageSet *** *** *** *** *** *** *** *** *** *** */
+
```

Comment:
Instead of several `cImageSet*` classes, it might be useful to consider using nested classes instead. Repeating "cImageSet" all the time in the class names seems useless, more information is conved by using `cImageSet::Surface`, etc.

That’s just my opinion, though.

Vale,
Quintus